service/qos/service_level_controller.hh

```diff
@@ -118,6 +118,7 @@ private:
     static constexpr shard_id global_controller = 0;
 
     std::map<sstring, service_level> _service_levels_db;
+    std::map<sstring, service_level_options> _effective_service_levels_db;
```

Comment:
The term "effective service level" is not too clear without reading an explanation, and here it's not too clear what the key is supposed to be. It's particularly confusing that `_effective_service_levels_db` looks very similar (the same key type, defined right next to it) to `_service_levels_db`, but the first one is indexed by _role_ name and the other is indexed by _service level_ name.

I'd appreciate a short comment around this variable explaining what the key is.